lib/composer.js

```diff
@@ -60,41 +63,43 @@ export default class Composer {
     this.destroyErrorIndicator()
     this.showProgressIndicator()
 
-    return new Promise(async (resolve, reject) => {
-      latex.log.group('LaTeX Build')
-      const rootFilePath = this.resolveRootFilePath(filePath)
-      const jobnames = builder.getJobNamesFromMagic(rootFilePath)
-
-      try {
-        for (const jobname of jobnames) {
-          const statusCode = await builder.run(rootFilePath, jobname)
-          const result = builder.parseLogAndFdbFiles(rootFilePath, jobname)
-
-          if (result) {
-            for (const message of result.messages) {
-              latex.log.showMessage(message)
-            }
-          }
-
-          if (statusCode > 0 || !result || !result.outputFilePath) {
-            this.showError(statusCode, result, builder)
-            reject(statusCode)
-          } else {
-            if (this.shouldMoveResult()) {
-              this.moveResult(result, rootFilePath)
-            }
-            this.showResult(result)
-          }
-        }
-      } catch (error) {
-        latex.log.error(error.message)
-        reject(error.message)
-      } finally {
+    latex.log.group('LaTeX Build')
+    const rootFilePath = this.resolveRootFilePath(filePath)
+    const jobnames = builder.getJobNamesFromMagic(rootFilePath)
+
+    if (this.rebuildCompleted && !this.rebuildCompleted.has(rootFilePath)) {
+      shouldRebuild = true
+      this.rebuildCompleted.add(rootFilePath)
+    }
+
+    const jobs = jobnames.map(jobname => this.buildJob(builder, rootFilePath, jobname, shouldRebuild))
+
+    return Promise.all(jobs)
```

Comment:
Does this make sense given that you're handling errors and successes in the containing function? I'm not 100% sure, but the only errors that might be raised on the calling side will originate from inside the `then` block. E.g. if either `this.destroyProgressIndicator()` or `latex.log.groupEnd()` results in an exception being raised.

Let's return `true/false`, or move the `then/catch` code to the call site?